ext/lmdb-safe/lmdb-safe.hh

```diff
@@ -12,9 +15,95 @@
 #include <mutex>
 #include <vector>
 #include <algorithm>
+#include <bit>
+#include "../../pdns/misc.hh"
 
 using std::string_view;
 
+#ifndef DNSDIST
+
+// FIXME do something more portable than __builtin_bswap64
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define _LMDB_SAFE_BSWAP64MAYBE(x) __builtin_bswap64(x)
+#else
+#define _LMDB_SAFE_BSWAP64MAYBE(x) (x)
+#endif
+
+namespace {
+  class __attribute__((__packed__)) LSheader {
+  public:
+    uint64_t d_timestamp;
+    uint64_t d_txnid;
+    uint8_t d_version;
+    uint8_t d_flags;
+    uint32_t d_reserved;
+    uint16_t d_numextra;
+
+    LSheader(uint64_t timestamp, uint64_t txnid, uint8_t flags=0, uint8_t version=0, uint8_t numextra=0):
+      d_timestamp(_LMDB_SAFE_BSWAP64MAYBE(timestamp)),
+      d_txnid(_LMDB_SAFE_BSWAP64MAYBE(txnid)),
+      d_version(version),
+      d_flags(flags),
+      d_reserved(0),
+      d_numextra(numextra)
```

Comment:
`numextra` also need to be stored in network order.

Passing it as an `uint8_t` does not save you here, because it is stored as `uint16_t` in the wrong byte order.

Reply:
ah, yes. I did have `ntohs` on the consuming side already.